Composer/packages/client/src/store/persistence/FileQueue.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { ResourceInfo, FileChangeType, IFileTask } from './types';
+
+export default class FileQueue {
```

Comment:
FileQueue looks like we are inventing new primitive types, usually something you don't do unless you have a strong reason

Reply:
Looks like this more like a TaskPool, at least to be a better name.

But in general, you don't need this taskPool, a 
var _taskQueues: {[key: string] : Task[] } should be good enough.

If you have more features, like a customized saving behavior or you are planning to move the executor into taskPool which is also often seen, then probably you need a taskPool class